### PR TITLE
[FLINK-18930][docs-zh] Translate "Hive Dialect" page of "Hive Integration" into Chinese

### PR DESCRIPTION
**Pull request by zhuxiaoshang**
## What is the purpose of the change

Translate "Hive Dialect" page of "Hive Integration" into Chinese

## Brief change log

- Translate "Hive Dialect" page of "Hive Integration" into Chinese

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (no)

### CODE REVIEW

**Comment by zhuxiaoshang** [comment]
cc @JingsongLi could you please have a review?

File: docs/dev/table/hive/hive_dialect.zh.md [hunk 1]
```diff
@@ -22,25 +22,19 @@ specific language governing permissions and limitations
...
-statement you execute. There's no need to restart a session to use a different dialect.
+Flink 目前支持两种 SQL 方言: `default` 和 `hive`。你需要先切换到 Hive 方言，然后才能使用 Hive 语法编写。下面介绍如何使用 SQL 客户端和 Table API 设置方言。
+还要注意，你可以为执行的每个语句动态切换方言。无需重新启动会话即可使用其他方言。
 
-### SQL Client
+### SQL 客户端
 
-SQL dialect can be specified via the `table.sql-dialect` property. Therefore you can set the initial dialect to use in
-the `configuration` section of the yaml file for your SQL Client.
+SQL 方言可以通过 `table.sql-dialect` 属性指定。因此你可以通过 `configuration` 章节中的 yaml 文件来为 SQL 客户端设置初始方言。
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_dialect.zh.md:37]:
```suggestion
SQL 方言可以通过 `table.sql-dialect` 属性指定。因此你可以通过 SQL 客户端 yaml 文件中的 `configuration` 部分来设置初始方言。
```

File: docs/dev/table/hive/hive_dialect.zh.md [hunk 2]
```diff
@@ -138,7 +131,7 @@ CREATE (DATABASE|SCHEMA) [IF NOT EXISTS] database_name
 ALTER (DATABASE|SCHEMA) database_name SET DBPROPERTIES (property_name=property_value, ...);
 {% endhighlight %}
 
-##### Update Owner
+##### Update Ower
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_dialect.zh.md:134]: typo

File: docs/dev/table/hive/hive_dialect.zh.md [hunk 3]
```diff
@@ -352,25 +342,18 @@ DROP FUNCTION [IF EXISTS] function_name;
...
-[Hive version]({{ site.baseurl }}/zh/dev/table/hive/#支持的hive版本) you use. For example, updating database
-location is only supported in Hive-2.4.0 or later.
-- Hive and Calcite have different sets of reserved keywords. For example, `default` is a reserved keyword in Calcite and
-a non-reserved keyword in Hive. Even with Hive dialect, you have to quote such keywords with backtick ( ` ) in order to
-use them as identifiers.
-- Due to expanded query incompatibility, views created in Flink cannot be queried in Hive.
+- Hive 方言只能用于操作 Hive 表，不能用于一般表。Hive 方言应与[HiveCatalog]({{ site.baseurl }}/zh/dev/table/hive/hive_catalog.html)一起使用。
+- 虽然所有 Hive 版本支持相同的语法，但是一些特定的功能是否可用仍取决于你使用的[Hive 版本]({{ site.baseurl }}/zh/dev/table/hive/#支持的hive版本)。例如，更新数据库位置
+ 只在 Hive-2.4.0 或更高版本支持。
+- Hive 和 Calcite 有不同的保留关键字集合。例如，`default` 是 Calcite 的保留关键字，却不是 Hive 的保留关键字。即使如此，在 Hive 方言中, 也必须使用反引号 ( ` ) 引用此类关键字才能将其用作标识符。
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_dialect.zh.md:358]:
```suggestion
- Hive 和 Calcite 有不同的保留关键字集合。例如，`default` 是 Calcite 的保留关键字，却不是 Hive 的保留关键字。即使使用 Hive 方言, 也必须使用反引号 ( ` ) 引用此类关键字才能将其用作标识符。
```

File: docs/dev/table/hive/hive_dialect.zh.md [hunk 4]
```diff
@@ -352,25 +342,18 @@ DROP FUNCTION [IF EXISTS] function_name;
...
+- 由于扩大了查询的不兼容性，在 Flink 中创建的视图是不能在 Hive 中查询的。
```
Comment (lirui-apache) [inline docs/dev/table/hive/hive_dialect.zh.md:359]:
```suggestion
- 由于扩展的查询语句的不兼容性，在 Flink 中创建的视图是不能在 Hive 中查询的。
```

**Comment by lirui-apache** [comment]
Thanks @zhuxiaoshang for working on the translation, nice work! I left some minor comments.

**Comment by zhuxiaoshang** [comment]
> Thanks @zhuxiaoshang for working on the translation, nice work! I left some minor comments.

Thank you @lirui-apache for your review,i'll fix it

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**